src/richie/apps/search/indexers/courses.py

```diff
@@ -183,9 +192,35 @@ def get_data_for_es(cls, index, action):
                     simple_text.body
                 )
 
+            # Make sure we get title information for categories in the same request
+            categories = (
+                course.get_root_to_leaf_category_pages()
+                .prefetch_related(
+                    Prefetch(
+                        "title_set",
+                        to_attr="published_titles",
+                        queryset=Title.objects.filter(published=True),
+                    )
+                )
+                .only("pk")
+            )
+
+            # Make sure we get title information for organizations in the same request
+            organizations = (
+                course.get_organizations()
+                .prefetch_related(
+                    Prefetch(
+                        "extended_object__title_set",
+                        to_attr="published_titles",
+                        queryset=Title.objects.filter(published=True),
+                    )
+                )
+                .only("public_extension", "extended_object")
```

Comment:
I think you can remove `public_extension` because the code below to get the list of organization ids is making another query anyway.